instrumentation/opentelemetry-instrumentation-starlette/pyproject.toml

```diff
@@ -35,7 +35,7 @@ dependencies = [
 
 [project.optional-dependencies]
 instruments = [
-  "starlette ~= 0.13.0",
+  "starlette ~= 0.13",
```

Comment:
@Kludex Alternatively, we could just do `starlette >= 0.13.0`. I'm not sure we can reasonable choose an upper bound, wdyt?

Reply:
0.13 is too old, some security vulnerabilities released since then... I'm not sure the exact version right now (on the phone), but at least 0.19.0 plus.

Reply:
Ah, I misread what you said.

I think we thought it was working... so I guess we can?

Reply:
So turns out the instrumentation fails tests on any recent versions of starlette. The newest I could get is `starlette >= 0.13, <0.15` 😞 

Mostly seeing errors with extracting custom headers:
`E   AssertionError: 'http.response.header.custom_test_header_1' not found in mappingproxy({'http.scheme': 'ws', 'http.host': 'testserver', 'net.host.port': 80, 'http.target': '/foobar_web', 'http.url': 'ws://testserver/foobar_web', 'http.server_name': 'testserver', 'http.user_agent': 'testclient', 'net.peer.ip': 'testclient', 'net.peer.port': 50000, 'http.route': '/foobar_web', 'http.status_code': 200})`

Reply:
I'd like to merge this PR as is, and I'll create a follow up issue for starlette

Reply:
https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3317